### PR TITLE
SAK-31048 Normalised announcement menu tabs

### PR DESCRIPTION
**Pull request by adrianfish**

### CODE REVIEW

File: velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java [hunk 1]
```diff
@@ -306,6 +309,15 @@ public boolean getIsField()
...
+		return m_current;
+	}
+
+	public void setIsCurrent(boolean current) {
+		m_current = current;
+	}
+
 	@Override
 	public String getAccessibilityLabel() {
 		return ((m_accessibilitylabel == null) ? m_title : m_accessibilitylabel);
```
Comment (duchuyctlk) [inline velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java]: The outermost parentheses are redundant.

Comment (adrianfish) [inline velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java]: Do you mean like this:

public void setIsCurrent(boolean current) m_current = current;

?

Comment (adrianfish) [inline velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java]: What do you mean?

Comment (ottenhoff) [inline velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java]: The comment is about code you did not modify. Please ignore code that is not part of the PR.

Comment (adrianfish) [inline velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java]: @duchuyctlk Ah, you mean the ternary line.

**Comment by adrianfish** [comment]
I've just squashed a comment on. That's it.

File: announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java [hunk 2]
```diff
@@ -1426,16 +1426,7 @@ private boolean isOkayToDisplayMessageMenu(AnnouncementActionState state)
...
-		{
-			return (statusName == null || statusName.equals(CANCEL_STATUS) || statusName.equals(POST_STATUS)
-					|| statusName.equals(DELETE_ANNOUNCEMENT_STATUS) || statusName.equals(FINISH_DELETING_STATUS)
-					|| statusName.equals("noSelectedForDeletion") || statusName.equals(NOT_SELECTED_FOR_REVISE_STATUS));
-		}
-		else
-		{
-			return false;
-		}
+		return SiteService.allowUpdateSite(ToolManager.getCurrentPlacement().getContext()) && !isOnWorkspaceTab();
```
Comment (ottenhoff) [inline announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java]: Can you briefly explain why the 10 lines of code above can be consolidated down to one line?

Comment (adrianfish) [inline announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java]: Yep. You always want the tabs to be showing, as long as you have the appropriate permission. You shouldn't have tabs popping into and out of existence depending on what view state you are in - they should just be fixed with just current tab being indicated. All those STATUS things are view states.

File: velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java (hunk 1 above)
Comment (duchuyctlk) [inline velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java]: Yes I meant the parentheses outside of the ternary , please sorry for my late reply.

Comment (relong76) [inline velocity/tool/src/java/org/sakaiproject/cheftool/menu/MenuEntry.java]: That line isn't part of the pull request... I'd stick to just commenting on the actual lines in the PR.
